include/aws/http/private/h2_decoder.h

```diff
@@ -27,10 +27,14 @@ struct aws_h2_decoder_vtable {
 
     int (*on_rst_stream)(uint32_t stream_id, uint32_t error_code, void *userdata);
 
-    int (*on_ping)(bool ack, uint64_t opaque_data, void *userdata);
+    int (*on_custom_frame_begin)(uint32_t stream_id, uint32_t payload_length, void *userdata);
+    int (*on_custom_frame_payload)(uint32_t stream_id, const struct aws_byte_cursor *data, void *userdata);
```

Comment:
I would do what the spec says and simply "ignore and discard" any unrecognized data.

It's not for people to introduce their own new custom data frames, they can use DATA for that. Any new frame type would likely be something intended to alter the state of the connection, and an external user can't do that.